assets/blocks/core-image/index.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Spinner } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import './style.scss';
+
+/**
+ * Add image credit meta to core/image block attributes
+ */
+addFilter(
+	'blocks.registerBlockType',
+	'newspack-plugin/register-hook/core-image',
+	( settings, name ) => {
+		if ( name !== 'core/image' ) {
+			return settings;
+		}
+		return {
+			...settings,
+			attributes: {
+				...settings.attributes,
+				meta: {
+					type: 'object',
+					default: {},
+				},
+			},
+		};
+	}
+);
+
+/**
+ * Display spinner and lock post saving until meta attributes are added to block
+ */
+const AttributesLoader = props => {
+	const imageId = props.attributes.id;
+	const { lockPostSaving, unlockPostSaving } = useDispatch( 'core/editor' );
+	const { meta } = useSelect( select => select( 'core' ).getMedia( imageId ), [ imageId ] ) ?? {
+		meta: {},
+	};
+	useEffect( () => {
+		// Meta added, proceed
+		if ( Object.keys( meta ).length ) {
+			props.setAttributes( { meta } );
+			unlockPostSaving( 'attachment-meta-empty' );
+		} else {
+			lockPostSaving( 'attachment-meta-empty' );
+		}
+	}, [ Object.keys( meta ).length ] );
+
+	return (
+		<>
+			{ ! Object.keys( meta ).length && (
+				<div className="newspack-block__core-image-background">
+					<div className="newspack-block__core-image-spinner">
+						<Spinner />
+					</div>
+				</div>
+			) }
+		</>
+	);
+};
+
+/**
+ * Helper to parse credit meta
+ *
+ * @param {{}} credit Credit meta to parse
+ * @return {string} Formatted string of image credit
+ */
+function parseCreditToText( credit ) {
+	const parsed = {
+		credit: credit._media_credit ?? '',
+		url: credit._media_credit_url ?? '',
+		org: credit._navis_media_credit_org ?? '',
+	};
+	if ( ! parsed.credit ) {
+		return '';
+	}
+	const org = `${ parsed.org ? ` / ${ parsed.org }` : '' }`;
+	return `${ __( 'Credit', 'newspack-plugin' ) }: ${ parsed.credit }${ org }`;
+}
+
+/**
+ * Populate attributes with meta data.
+ */
+addFilter(
+	'editor.BlockEdit',
+	'newspack-plugin/block-edit-hook/core-image',
+	createHigherOrderComponent( BlockEdit => {
+		const blockEditComponent = props => {
+			if ( props.name === 'core/image' ) {
+				return (
+					<div
+						className="newspack-block__core-image"
+						style={ {
+							'--wp-block-image-caption-credit': `" ${ parseCreditToText(
```

Comment:
This will only work if the "Caption" block attribute is toggled on: 

<img width="302" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/009635a0-e7fb-4b13-878a-0ddb489927cb">

Feel free to drop this since as you've mentioned it's not part of requirements. Otherwise, the "Caption" attribute should be programmatically toggled on as part of these changes.

Reply:
@adekbadek I have recreated the caption mechanism/components to mimic current behavior + include credits.
Look forward to your feedback

Reply:
I appreciate the effort, but recreating the whole toolbar is overcomplicating the code and not solving the issue, unfortunately 😬  With these changes, the credit will be visible on the front-end regardless of caption, but in the editor it will be visible only if the caption is toggled on. What I meant in my suggestion was that the  block attribute should be uneditable if the credit is set – this way the presentation is the same in the editor and on the front-end.

Reply:
@adekbadek thanks for the feedback. The issue (the premise of this PR) was to ensure credits display on distributed (using 10up/distributor) posts. This is done and works 100%, if you feel differently please share your feedback concerning this specifically.

Since the issue was resolved we've moved onto your recommendation i.e. "WYSIWYG". Let me try and explain what was necessary to achieve this.

Current functionality:
* Frontend: If credit exists, it will always display
* Frontend: If caption exists and has been enabled on block, it will always display
* Editor: `image/core` displays a caption if it exists and is toggled on
* Editor: No credit is displayed

The challenge:
`core/image` is a static and dynamic block. Captions are static and since we're unable to provide a `deprecation` mechanism the only way to manipulate how image captions display/behave in the editor is by recreating them and applying customizations to that. I'm unaware of another approach?

The last piece which I discovered this morning was to clear the caption when toggled-off. 1646f7c0bdf51be674e613d19e11f1ca0bd7a11f fixes that.

Reply:
> The issue (the premise of this PR) was to ensure credits display on distributed (using 10up/distributor) posts. This is done and works 100%

Yep, that's all good 👌 

> The challenge

I'm aware that this is challenging and that's why I've proposed to drop the the editor part of this PR. As it currently is, it's not WYSIWYG – the display of the credit is tied to the display of the caption, which is not true for the front-end. This is confusing for the user, less so than the credit appearing on the front-end while missing in the editor.  

The changes to the front-end are enough to satisfy requirements, so if the editor UI part is removed then we're good to go here.

Reply:
>the display of the credit is tied to the display of the caption, which is not true for the front-end.

With the latest code _this_ is current behavior.

**Has caption and credit:**

(Frontend)
![Screenshot 2024-02-21 at 10 19 23](https://github.com/Automattic/newspack-plugin/assets/3676975/78041ab2-7138-4c0c-90f2-43432ef9c7ac)

(Backend)
![Screenshot 2024-02-21 at 10 20 30](https://github.com/Automattic/newspack-plugin/assets/3676975/17a88fed-837d-4ad5-bc1e-b6c05537006e)

**No caption, has credit**

(Frontend)
![Screenshot 2024-02-21 at 10 21 33](https://github.com/Automattic/newspack-plugin/assets/3676975/cbd3f4b2-b446-4fc8-807b-db4d46aa18f5)

(Backend)
![Screenshot 2024-02-21 at 10 20 49](https://github.com/Automattic/newspack-plugin/assets/3676975/7da0f8ab-34ed-4171-afe1-9bd63fbe1830)

Let me know if you would still like me to remove "editor" parts.

Reply:
> With the latest code this is current behavior.

Confirmed! 1646f7c0bdf51be674e613d19e11f1ca0bd7a11f fixed the issue I was describing.